sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java

```diff
@@ -140,6 +143,7 @@ private InterceptorManager(String testName, String playbackRecordName, TestMode
         } else {
             this.recordedData = null;
         }
+        this.recordSanitizers = new ArrayList<>();
```

Comment:
I would either move this to the condition that the test proxy is enabled or initialize it in the class and mark it `final`.